server/src/main/java/org/elasticsearch/index/query/QueryShardContext.java

```diff
@@ -92,12 +93,35 @@
 
     private final Index fullyQualifiedIndex;
     private final Predicate<String> indexNameMatcher;
+    private final BooleanSupplier isAllowExpensiveQueries;
 
     private final Map<String, Query> namedQueries = new HashMap<>();
     private boolean allowUnmappedFields;
     private boolean mapUnmappedFieldAsString;
     private NestedScope nestedScope;
 
+    public QueryShardContext(int shardId,
+                             IndexSettings indexSettings,
+                             BigArrays bigArrays,
+                             BitsetFilterCache bitsetFilterCache,
+                             BiFunction<MappedFieldType, String, IndexFieldData<?>> indexFieldDataLookup,
+                             MapperService mapperService,
+                             SimilarityService similarityService,
+                             ScriptService scriptService,
+                             NamedXContentRegistry xContentRegistry,
+                             NamedWriteableRegistry namedWriteableRegistry,
+                             Client client,
+                             IndexSearcher searcher,
+                             LongSupplier nowInMillis,
+                             String clusterAlias,
+                             Predicate<String> indexNameMatcher,
+                             BooleanSupplier isAllowExpensiveQueries) {
+        this(shardId, indexSettings, bigArrays, bitsetFilterCache, indexFieldDataLookup, mapperService, similarityService,
+                scriptService, xContentRegistry, namedWriteableRegistry, client, searcher, nowInMillis, indexNameMatcher,
+                new Index(RemoteClusterAware.buildRemoteIndexName(clusterAlias, indexSettings.getIndex().getName()),
+                        indexSettings.getIndex().getUUID()), isAllowExpensiveQueries);
+    }
```

Comment:
can we avoid duplicating constructors?

Reply:
There are ~20 usages of this in tests, so no big deal, I can remove the constructor.